packages/dds/test-dds-utils/src/ddsSnapshotHarness.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+/**
+ * @internal
+ */
+export interface ISnapshotSuite {
+	/**
+	 * A utility function for setting up the current snapshot directory for tests.
+	 */
+	useSnapshotDirectory: () => void;
+
+	/**
+	 * Takes a snapshot of data and writes it to a file.
+	 * @param data - The data to take a snapshot of.
+	 * @param writeCompatible - A flag indicating whether the snapshot should be checked for consistency
```

Comment:
let's keep `writeCompatible`-ness out of this layer. The API here should mimic that of `packages\dds\tree\src\test\snapshots\snapshotTools.ts` as closely as possible. The main difference is adding a single layer of indirection for the base folder to put the path (which is what `createSnapshotSuite` is). So you still want a `string` parameter on `useSnapshotDirectory`, but since it has subdirectory-like semantics i'd instead call it `useSnapshotSubdirectory`.